osu.Game/OsuGame.cs

```diff
@@ -1093,6 +1091,24 @@ private void screenPushed(IScreen lastScreen, IScreen newScreen)
         {
             ScreenChanged(lastScreen, newScreen);
             Logger.Log($"Screen changed → {newScreen}");
+
+            // set AllowTrackAdjustments if new screen defines it, inherit otherwise
+            if (newScreen is IOsuScreen newOsuScreen && newOsuScreen.AllowTrackAdjustments.HasValue)
+            {
+                allowTrackAdjustmentsStack.Push(newOsuScreen.AllowTrackAdjustments.Value);
+                Logger.Log($"Screen's AllowTrackAdjustments explicit → {allowTrackAdjustmentsStack.First()}");
+            }
+            else if (allowTrackAdjustmentsStack.Any())
+            {
+                allowTrackAdjustmentsStack.Push(allowTrackAdjustmentsStack.First());
```

Comment:
all `.First()` calls should probably be `.Peek()` calls (more obvious to me, as at least I have to think what `.First()` means for a stack)